src/ShopCertification.php

```diff
@@ -117,11 +118,20 @@ public function setOrderId($orderId)
     /**
      * @param string $productItemId Must be same as ITEM_ID provided in the Heureka XML feed.
      *
-     * @return self
+     * @return ShopCertification
+     * @throws DuplicateProductItemIdException
      */
     public function addProductItemId($productItemId)
     {
-        $this->productItemIds[] = (string)$productItemId;
+        $productItemId = (string)$productItemId;
+
+        if (isset($this->productItemIds[$productItemId])) {
```

Comment:
Neviděl jsem e-shop, který by měl nakupování více kusů řešeno přes duplikací položek v košíku. Fakt si myslíte, že to může nastat? Protože mě se docela tohle řešení líbí (když to někdo posere, tak to uvidí ...)